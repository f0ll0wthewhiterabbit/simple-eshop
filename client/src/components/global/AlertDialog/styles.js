import styled from 'styled-components'
import { Dialog } from '@material-ui/core'

export const DialogContainer = styled(Dialog)`
  .MuiDialog-paper {
    padding-top: ${props => props.theme.spacing(1)}px;
    padding-bottom: ${props => props.theme.spacing(1)}px;
    background-color: ${props => props.theme.colors.background.alertDialog};
  }

  .MuiDialogTitle-root h2 {
    color: ${props => props.theme.colors.font.bold};
  }

  .MuiDialogContent-root p {
    color: ${props => props.theme.colors.font.regular};
  }

  .MuiDialogActions-root button:first-child {
    background-color: #fff;
  }
`
