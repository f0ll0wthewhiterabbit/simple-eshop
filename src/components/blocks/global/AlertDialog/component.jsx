import React from 'react'
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'

const AlertDialog = ({ title, children, show, handleModalClose }) => {
  return (
    <Dialog
      open={show}
      onClose={handleModalClose}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
      {children && (
        <DialogContent>
          <DialogContentText id="alert-dialog-description">{children}</DialogContentText>
        </DialogContent>
      )}
      <DialogActions>
        <Button onClick={handleModalClose} color="primary">
          Cancel
        </Button>
        <Button onClick={handleModalClose} color="primary" autoFocus>
          Confirm
        </Button>
      </DialogActions>
    </Dialog>
  )
}

AlertDialog.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  show: PropTypes.string.isRequired,
  handleModalClose: PropTypes.func.isRequired,
}

export default AlertDialog
