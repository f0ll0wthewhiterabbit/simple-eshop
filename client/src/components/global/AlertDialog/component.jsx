import React from 'react'
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'
import ErrorIcon from '@material-ui/icons/Error'

import {
  STORE_FIELD_PRODUCTS,
  STORE_FIELD_USERS,
  STORE_FIELD_CURRENT_USER,
} from '../../../constants'
import { DialogContainer } from './styles'

const AlertDialog = ({
  isModalOpened,
  title,
  children,
  closeModal,
  storeFieldName,
  deleteUsers,
  deleteProducts,
  requestUserDeletion,
}) => {
  const handleConfirmButton = () => {
    switch (storeFieldName) {
      case STORE_FIELD_PRODUCTS:
        deleteProducts()
        break

      case STORE_FIELD_USERS:
        deleteUsers()
        break

      case STORE_FIELD_CURRENT_USER:
        requestUserDeletion()
        break

      default:
        break
    }
  }

  return (
    <DialogContainer
      open={isModalOpened}
      onClose={closeModal}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
      data-test="dialogContainer"
    >
      <DialogTitle id="alert-dialog-title" data-test="dialogTitle">
        {title}
      </DialogTitle>
      {children && (
        <DialogContent>
          <DialogContentText id="alert-dialog-description" data-test="dialogContentText">
            {children}
          </DialogContentText>
        </DialogContent>
      )}
      <DialogActions>
        <Button onClick={closeModal} color="primary" variant="outlined" data-test="cancelButton">
          Cancel
        </Button>
        <Button
          onClick={handleConfirmButton}
          color="secondary"
          variant="contained"
          startIcon={<ErrorIcon />}
          autoFocus
          data-test="confirmButton"
        >
          Confirm
        </Button>
      </DialogActions>
    </DialogContainer>
  )
}

AlertDialog.defaultProps = {
  children: '',
}

AlertDialog.propTypes = {
  isModalOpened: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  closeModal: PropTypes.func.isRequired,
  storeFieldName: PropTypes.string.isRequired,
  deleteUsers: PropTypes.func.isRequired,
  deleteProducts: PropTypes.func.isRequired,
  requestUserDeletion: PropTypes.func.isRequired,
}

export default AlertDialog
