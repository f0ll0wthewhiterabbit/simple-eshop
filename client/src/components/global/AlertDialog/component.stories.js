import React from 'react'
import { action } from '@storybook/addon-actions'

import AlertDialog from './component'

export default {
  title: 'AlertDialog',
  component: AlertDialog,
  excludeStories: /.*Data$/,
}

export const alertDialogData = {
  isModalOpened: true,
  title: 'Test title',
  childrenMessage: `Lorem ipsum dolor, sit amet consectetur adipisicing elit.
  Consectetur quae omnis sunt voluptatum sequi quos ducimus soluta quasi ipsum suscipit!
  Fugit, facilis? Voluptates, et? Et adipisci labore pariatur in laborum.`,
}

const { isModalOpened, title, childrenMessage } = alertDialogData

export const actionsData = {
  closeModal: action('closeModal'),
  confirmMethod: action('confirmMethod'),
}

export const normal = () => (
  <AlertDialog isModalOpened={isModalOpened} title={title} {...actionsData}>
    {childrenMessage}
  </AlertDialog>
)
export const withoutMessage = () => (
  <AlertDialog isModalOpened={isModalOpened} title={title} {...actionsData} />
)
export const withMessageAsNode = () => (
  <AlertDialog isModalOpened={isModalOpened} title={title} {...actionsData}>
    <i>{childrenMessage}</i>
    <ul>
      <li>Fitst</li>
      <li>Second</li>
    </ul>
  </AlertDialog>
)
