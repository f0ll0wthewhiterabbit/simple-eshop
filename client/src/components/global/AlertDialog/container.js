import { connect } from 'react-redux'

import AlertDialog from './component'
import { closeModal, deleteUsers, deleteProducts, requestUserDeletion } from '../../../store/actions'

const mapStateToProps = state => ({
  isModalOpened: state.app.modal.isOpened,
  storeFieldName: state.app.modal.storeFieldName,
})

const mapDispatchToProps = dispatch => ({
  closeModal: () => dispatch(closeModal()),
  deleteUsers: () => dispatch(deleteUsers()),
  deleteProducts: () => dispatch(deleteProducts()),
  requestUserDeletion: () => dispatch(requestUserDeletion()),
})

export default connect(mapStateToProps, mapDispatchToProps)(AlertDialog)
