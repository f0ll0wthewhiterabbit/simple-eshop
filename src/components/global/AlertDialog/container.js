import { connect } from 'react-redux'

import AlertDialog from './component'
import { closeModal, deleteSelectedItems } from '../../../store/actions'

const mapStateToProps = state => ({
  isModalOpened: state.app.modal.isOpened,
  storeFieldName: state.app.modal.storeFieldName,
})

const mapDispatchToProps = dispatch => ({
  closeModal: () => dispatch(closeModal()),
  deleteSelectedItems: storeFieldName => dispatch(deleteSelectedItems(storeFieldName)),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AlertDialog)
