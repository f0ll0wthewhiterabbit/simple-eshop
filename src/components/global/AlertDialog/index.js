import AlertDialog from './component'

export default AlertDialog
