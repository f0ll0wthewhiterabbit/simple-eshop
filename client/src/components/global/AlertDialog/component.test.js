import React from 'react'
import { shallow } from 'enzyme'

import AlertDialog from './component'
import findByTestAttr from '../../../utils/findByTestAttr'
import {
  STORE_FIELD_PRODUCTS,
  STORE_FIELD_USERS,
  STORE_FIELD_CURRENT_USER,
} from '../../../constants'

describe('AlertDialog component', () => {
  let props
  let wrapper
  const testTitle = 'Test Title'
  const mockCloseModal = jest.fn()
  const mockDeleteUsers = jest.fn()
  const mockDeleteProducts = jest.fn()
  const mockRequestUserDeletion = jest.fn()

  const generateWrapper = passedProps => {
    const initialProps = {
      isModalOpened: false,
      title: testTitle,
      children: '',
      closeModal: mockCloseModal,
      storeFieldName: '',
      deleteUsers: mockDeleteUsers,
      deleteProducts: mockDeleteProducts,
      requestUserDeletion: mockRequestUserDeletion,
    }
    props = { ...initialProps, ...passedProps }

    return shallow(<AlertDialog {...props} />)
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should be closed', () => {
    const dialogContainer = findByTestAttr(wrapper, 'dialogContainer')

    expect(dialogContainer.props().open).toBe(false)
  })

  it(`should be opened if prop 'isModalOpened=true'`, () => {
    wrapper = generateWrapper({ isModalOpened: true })
    const dialogContainer = findByTestAttr(wrapper, 'dialogContainer')

    expect(dialogContainer.props().open).toBe(true)
  })

  it('should contain correct title', () => {
    const dialogTitle = findByTestAttr(wrapper, 'dialogTitle')

    expect(dialogTitle.text()).toBe(testTitle)
  })

  it(`shouldn't contain additional text`, () => {
    const dialogContentText = findByTestAttr(wrapper, 'dialogContentText')

    expect(dialogContentText).toHaveLength(0)
  })

  it('should contain additional text passed as children', () => {
    const testAdditionalText = 'test additional text'
    wrapper = generateWrapper({ children: testAdditionalText })
    const dialogContentText = findByTestAttr(wrapper, 'dialogContentText')

    expect(dialogContentText).toHaveLength(1)
    expect(dialogContentText.text()).toBe(testAdditionalText)
  })

  it('should be closed after close button click', () => {
    const cancelButton = findByTestAttr(wrapper, 'cancelButton')
    cancelButton.simulate('click')

    expect(mockCloseModal).toHaveBeenCalledTimes(1)
  })

  it('should be confirmed but not handled after confirm button click', () => {
    const confirmButton = findByTestAttr(wrapper, 'confirmButton')
    confirmButton.simulate('click')

    expect(mockDeleteUsers).toHaveBeenCalledTimes(0)
    expect(mockDeleteProducts).toHaveBeenCalledTimes(0)
    expect(mockRequestUserDeletion).toHaveBeenCalledTimes(0)
  })

  it('should handle delete users after confirm button click', () => {
    wrapper = generateWrapper({ storeFieldName: STORE_FIELD_USERS })
    const confirmButton = findByTestAttr(wrapper, 'confirmButton')
    confirmButton.simulate('click')

    expect(mockDeleteUsers).toHaveBeenCalledTimes(1)
  })

  it('should handle delete products after confirm button click', () => {
    wrapper = generateWrapper({ storeFieldName: STORE_FIELD_PRODUCTS })
    const confirmButton = findByTestAttr(wrapper, 'confirmButton')
    confirmButton.simulate('click')

    expect(mockDeleteProducts).toHaveBeenCalledTimes(1)
  })

  it('should handle deletetion request of user after confirm button click', () => {
    wrapper = generateWrapper({ storeFieldName: STORE_FIELD_CURRENT_USER })
    const confirmButton = findByTestAttr(wrapper, 'confirmButton')
    confirmButton.simulate('click')

    expect(mockRequestUserDeletion).toHaveBeenCalledTimes(1)
  })
})
