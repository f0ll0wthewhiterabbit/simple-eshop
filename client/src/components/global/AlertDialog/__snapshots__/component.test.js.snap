// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialog component should render correctly 1`] = `
<Styled(WithStyles(ForwardRef(Dialog)))
  aria-describedby="alert-dialog-description"
  aria-labelledby="alert-dialog-title"
  data-test="dialogContainer"
  onClose={[MockFunction]}
  open={false}
>
  <WithStyles(ForwardRef(DialogTitle))
    data-test="dialogTitle"
    id="alert-dialog-title"
  >
    Test Title
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      data-test="cancelButton"
      onClick={[MockFunction]}
      variant="outlined"
    >
      Cancel
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      autoFocus={true}
      color="secondary"
      data-test="confirmButton"
      onClick={[Function]}
      startIcon={<Memo(ErrorIcon) />}
      variant="contained"
    >
      Confirm
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</Styled(WithStyles(ForwardRef(Dialog)))>
`;
